AmplifyPlugins/DataStore/Tests/AWSDataStorePluginTests/Core/AWSDataStorePluginTests.swift

```diff
@@ -90,6 +90,75 @@ class AWSDataStorePluginTests: XCTestCase {
+        let startExpectation = expectation(description: "Start Sync should be called")
+        var currCount = 0
+        let storageEngine = MockStorageEngineBehavior()
+        
+        storageEngine.responders[.startSync] = StartSyncResponder { _ in
+            currCount = self.expect(startExpectation, currCount, 1)
+        }
+        
+        let storageEngineBehaviorFactory: StorageEngineBehaviorFactory = {_, _, _, _, _, _  throws in
+            return storageEngine
+        }
+        let dataStorePublisher = DataStorePublisher()
+        let plugin = AWSDataStorePlugin(modelRegistration: TestModelRegistration(),
+                                        storageEngineBehaviorFactory: storageEngineBehaviorFactory,
+                                        dataStorePublisher: dataStorePublisher,
+                                        validAPIPluginKey: "MockAPICategoryPlugin",
+                                        validAuthPluginKey: "MockAuthCategoryPlugin")
+        
+        do {
+            try plugin.configure(using: nil)
+            XCTAssertNil(plugin.storageEngine)
+            try await plugin.stop()
+            await stopExpectation.fulfill()
+            
+            plugin.start(completion: { _ in })
+        } catch {
+            XCTFail("DataStore configuration should not fail with nil configuration. \(error)")
+        }
+        await waitForExpectations(timeout: 1.0)
+    }
+    
+    func testStorageEngineStartsOnPluginClearStart() async throws {
+        let clearExpectation = expectation(description: "Clear should be called")
+        let startExpectation = expectation(description: "Start Sync should be called")
+        var currCount = 0
+        
+        let storageEngine = MockStorageEngineBehavior()
+        storageEngine.responders[.clear] = ClearResponder { _ in
+            currCount = self.expect(clearExpectation, currCount, 1)
+        }
+        
+        storageEngine.responders[.startSync] = StartSyncResponder { _ in
+            currCount = self.expect(startExpectation, currCount, 2)
+        }
+        
+        let storageEngineBehaviorFactory: StorageEngineBehaviorFactory = {_, _, _, _, _, _  throws in
+            return storageEngine
+        }
+        let dataStorePublisher = DataStorePublisher()
+        let plugin = AWSDataStorePlugin(modelRegistration: TestModelRegistration(),
+                                        storageEngineBehaviorFactory: storageEngineBehaviorFactory,
+                                        dataStorePublisher: dataStorePublisher,
+                                        validAPIPluginKey: "MockAPICategoryPlugin",
+                                        validAuthPluginKey: "MockAuthCategoryPlugin")
+        
+        do {
+            try plugin.configure(using: nil)
+            XCTAssertNil(plugin.storageEngine)
+            
+            try await plugin.clear()
+            try await plugin.start()
```

Comment:
the PR's should be identical where possible such that the unit test is the same, we should be testing the callback APIs because the problem we're solving exists on `main` (v2) as well. Even though the `main` v2 callback APIs are not exposed to the developer through `Amplify.DataStore`, we should still test the implementation and fix for the bug works as expected